### PR TITLE
Prepare release 2.4

### PR DESCRIPTION
**Pull request by fpetkovski**
**What this PR does / why we need it**:
Prepares the 2.4 release.

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*
Does not change.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Review by liangyuanpeng — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -68,10 +68,10 @@ At most, 5 kube-state-metrics and 5 [kubernetes releases](https://github.com/kub
 
 | kube-state-metrics | **Kubernetes 1.19** | **Kubernetes 1.20** | **Kubernetes 1.21** | **Kubernetes 1.22** | **Kubernetes 1.23** |
 |--------------------|:-------------------:|:-------------------:|:-------------------:|:-------------------:|:-------------------:|
-| **v1.9.8**         |          -          |          -          |          -          |          -          |          -          |
 | **v2.1.1**         |          ✓          |          ✓          |          ✓          |         -/✓         |         -/✓         |
 | **v2.2.4**         |          ✓          |          ✓          |          ✓          |          ✓          |          ✓          |
 | **v2.3.0**         |          ✓          |          ✓          |          ✓          |          ✓          |          ✓          |
+| **v2.4.0**         |         -/✓         |         -/✓         |          ✓          |          ✓          |          ✓          |
```
Comment (dgrisonnet) [inline README.md:74]: This should be the same as master since we are cutting the release based on the master branch

Comment (fpetkovski) [reply README.md]: Good point, updated

**Pull request merged**